test/e2e/antreapolicy_test.go

```diff
@@ -2410,6 +2410,93 @@ func testFQDNPolicy(t *testing.T) {
 	time.Sleep(networkPolicyDelay)
 }
 
+// testFQDNPolicyInClusterService uses in-cluster headless Services to test a FQDN
+// policy to avoid a dependency on external connectivity. The reason we use
+// headless Service is that FQDN will use the IP from DNS record to implement flows
+// in egress table. For a not headless Service, the DNS name resolves to the
+// ClusterIP for the Service. But when traffic arrive egress table, the dstIP has
+// been DNAT to the Endpoints IP by AntreaProxy Service Load-Balancing. For
+// headless Service, the Endpoints IP will be directly returned by DNS. In this
+// case, FQDN based policies can be used successfully.
```

Comment:
suggested rewrite:

// testFQDNPolicyInClusterService uses in-cluster headless Services to test FQDN
// policies, to avoid having a dependency on external connectivity. The reason we use
// headless Service is that FQDN will use the IP from DNS A/AAAA records to implement flows
// in the egress policy table. For a non-headless Service, the DNS name resolves to the
// ClusterIP for the Service. But when traffic arrives to the egress table, the dstIP has
// already been DNATed to the Endpoints IP by AntreaProxy Service Load-Balancing, and
// the policies are not enforced correctly. For a headless Service, the Endpoints IP
// will be directly returned by the DNS server. In this case, FQDN based policies can be
// enforced successfully.

same for commit message

Reply:
Thanks. Done.